frame/scheduler/src/lib.rs

```diff
@@ -125,8 +168,37 @@ decl_error! {
 decl_module! {
 	/// Scheduler module declaration.
 	pub struct Module<T: Trait> for enum Call where origin: <T as system::Trait>::Origin {
+		type Error = Error<T>;
 		fn deposit_event() = default;
 
+		fn on_runtime_upgrade() -> Weight {
+			if StorageVersion::get() == Releases::V1 {
+				StorageVersion::put(Releases::V2);
+
+				Agenda::<T>::translate::<
+					Vec<Option<ScheduledV1<<T as Trait>::Call, T::BlockNumber>>>, _
+				>(|_, agenda| {
+					Some(
+						agenda
+						.into_iter()
+						.map(|schedule| schedule.map(|schedule| ScheduledV2 {
+							maybe_id: schedule.maybe_id,
+							priority: schedule.priority,
+							call: schedule.call,
+							maybe_periodic: schedule.maybe_periodic,
+							origin: system::RawOrigin::Root.into(),
+							_phantom: Default::default(),
+						}))
+						.collect::<Vec<_>>()
+					)
+				});
```

Comment:
This is the logic to export for use by the runtime's `lib.rs`. Each runtime (polkadot, kusama, westend) would need to introduce its own `on_runtime_upgrade` handler.

Reply:
I have abstracted the migration logic to a public method but unsure what changes is needed for `lib.rs`, any examples?

Reply:
@bkchr I'm also unfamiliar with the specifics of this. Do you have any code to point at for it?

Reply:
Here is the relevant pr: https://github.com/paritytech/substrate/pull/5782/files